app/bundles/CoreBundle/Doctrine/Type/UTCDateTimeType.php

```diff
@@ -69,7 +69,9 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
 
         // Set to local timezone
         date_default_timezone_set($timezone);
-        $value->setTimezone(new \DateTimeZone($timezone));
+        if ($value instanceof \DateTime) {
```

Comment:
@escopecz this is required to make sure the type is correct

Reply:
It would work for `DateTimeImmutable` too. We'd have to store the new version of the date to the same variable though. Since we expect `DateTimeInterface` then we should make that work too.

https://www.php.net/manual/en/datetimeimmutable.settimezone.php

Reply:
resolved in commit 68e481883c80465cab93ccfae54e4f7784b9e6f7